clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -989,6 +989,44 @@ def CmpOp : CIR_Op<"cmp", [Pure, SameTypeOperands]> {
   let hasVerifier = 0;
 }
 
+//===----------------------------------------------------------------------===//
+// BitsOp
+//===----------------------------------------------------------------------===//
+
+def BitsOpKind_CLRSB : I32EnumAttrCase<"clrsb", 1>;
+def BitsOpKind_CTZ : I32EnumAttrCase<"ctz", 2>;
+def BitsOpKind_CLZ : I32EnumAttrCase<"clz", 3>;
+def BitsOpKind_FFS : I32EnumAttrCase<"ffs", 4>;
+def BitsOpKind_PARITY : I32EnumAttrCase<"parity", 5>;
+def BitsOpKind_POPCOUNT : I32EnumAttrCase<"popcount", 6>;
+
+def BitsOpKind : I32EnumAttr<
+    "BitsOpKind", "bits op kind", [
+        BitsOpKind_CLRSB, BitsOpKind_CTZ, BitsOpKind_CLZ, BitsOpKind_FFS,
+        BitsOpKind_PARITY, BitsOpKind_POPCOUNT,
+    ]> {
+  let cppNamespace = "::mlir::cir";
+}
+
+def BitsOp : CIR_Op<"bits", [Pure]> {
```

Comment:
This is great, thanks!

I have two suggestions:
1. Use one op for each (so that verifiers can be dedicated) and avoid the enum kinds here. You can do it by having a generic CIROps_Bits class (see how `CallOp` and `TryCallOp` use a common `CIR_CallOp` operation). For example:
```
def BitCtzOp : CIR_BitOps<"ctz">;
...
```
2. Encode the group within the name: `cir.bit.ctz` (and use `bit` instead of `bits`).

Reply:
Actually I'm thinking we could prefix all these _builtin_ operations with `builtin`? For example, `cir.builtin.bit.ctz`, `cir.builtin.bit,clz`, etc. This may look tedious, though.

Reply:
I think we can skip the `builtin` part. IMO, from the user (C/C++ developer) perspective, these things indeed look like a compiler built-in, whereas for CIR, they are just operations that constitutes the whole of the IR, as much as other higher level operations we already support.